core/src/main/java/org/apache/iceberg/MergingSnapshotUpdate.java

```diff
@@ -243,19 +238,21 @@ protected void add(DataFile file) {
             group.add(manifest);
             groups.put(manifest.partitionSpecId(), group);
           }
-        }
+        });
       }
 
       List<ManifestFile> manifests = Lists.newArrayList();
-      for (Map.Entry<Integer, List<ManifestFile>> entry : groups.entrySet()) {
-        for (ManifestFile manifest : mergeGroup(entry.getKey(), entry.getValue())) {
-          manifests.add(manifest);
+      groups.forEach((groupId, manifestGroup) -> {
```

Comment:
Here's an example where I don't think it is a good trade-off to use the streams API.

Reply:
Let's resolve the discussion in https://github.com/apache/incubator-iceberg/pull/153#discussion_r274062236 and then apply whatever conclusion we draw everywhere.